xlml/utils/metric.py

```diff
@@ -322,19 +326,29 @@ def get_gcs_file_location_with_regex(file_location: str) -> str:
   file_path_regex = re.compile(file_path)
   prefix = "/".join(file_path.split("/")[:-1])
 
-  all_blobs_names = [
-      b.name for b in storage_client.list_blobs(bucket_name, prefix=prefix)
+  matched_blobs = [
+      b
+      for b in storage_client.list_blobs(bucket_name, prefix=prefix)
+      if file_path_regex.match(b.name)
   ]
 
-  try:
-    return (
-        f"gs://{bucket_name}/"
-        f"{next(filter(file_path_regex.match, all_blobs_names))}"
-    )
-  except StopIteration:
+  if not matched_blobs:
     logging.warning(f"No objects matched supplied regex: {file_location}")
     return ""
 
+  if return_largest:
```

Comment:
Trying to understand the logic of this largest file, i.e. should we process all files to get all metrics?

Reply:
Ideally yes, it'd be good to process all files, but probably as a future followup. Because:
1. For maxdiffusion-specific use case, the test always generates 2 tensorboard files, with the smaller one being a dummy (only 40 bytes). So picking the largest file works for our use case
2. metric.py currently is set up to expect only one tensorboard file, so some refactoring is required, which I think is better left as a separate PR.

Reply:
> the smaller one being a dummy

Not sure why, but sg in this case we extract largest file.